include/zephyr/bluetooth/conn.h

```diff
@@ -1774,6 +1774,66 @@ struct bt_br_conn_param {
 struct bt_conn *bt_conn_create_br(const bt_addr_t *peer,
 				  const struct bt_br_conn_param *param);
 
+/** @brief Life-span states of SCO channel. Used only by internal APIs
+ *  dealing with setting channel to proper state depending on operational
+ *  context.
+ */
+enum bt_sco_state {
+	/** Channel disconnected */
+	BT_SCO_STATE_DISCONNECTED,
+	/** Channel is pending ACL encryption before connecting */
+	BT_SCO_STATE_ENCRYPT_PENDING,
+	/** Channel in connecting state */
+	BT_SCO_STATE_CONNECTING,
+	/** Channel ready for upper layer traffic on it */
+	BT_SCO_STATE_CONNECTED,
+	/** Channel in disconnecting state */
+	BT_SCO_STATE_DISCONNECTING,
+};
+
+struct bt_sco_chan;
+struct bt_sco_chan_ops {
+	/** @brief Channel connected callback
+	 *
+	 *  If this callback is provided it will be called whenever the
+	 *  connection completes.
+	 *
+	 *  @param chan The channel that has been connected
+	 */
+	void (*connected)(struct bt_sco_chan *chan);
+
+	/** @brief Channel disconnected callback
+	 *
+	 *  If this callback is provided it will be called whenever the
+	 *  channel is disconnected, including when a connection gets
+	 *  rejected or when setting security fails.
+	 *
+	 *  @param chan   The channel that has been Disconnected
+	 *  @param reason BT_HCI_ERR_* reason for the disconnection.
+	 */
+	void (*disconnected)(struct bt_sco_chan *chan, uint8_t reason);
+};
+
+struct bt_sco_chan {
+	struct bt_conn *sco;
+	/** Channel operations reference */
+	struct bt_sco_chan_ops		*ops;
+
+	enum bt_sco_state		state;
+
+#if defined(CONFIG_BT_SMP) || defined(__DOXYGEN__)
+	/** @brief The required security level of the channel
+	 *
+	 * This value can be set as the central before calling bt_conn_create_sco().
+	 * The value is overwritten to @ref bt_iso_server::sec_level for the
+	 * peripheral once a channel has been accepted.
+	 *
+	 * Only available when @kconfig{CONFIG_BT_SMP} is enabled.
+	 */
+	bt_security_t			required_sec_level;
+#endif /* CONFIG_BT_SMP */
```

Comment:
Do we really need to track this independently from the parent ACL connection?

Reply:
Currently, security related checks are not necessary. I removed it.